source/common/http/conn_manager_impl.h

```diff
@@ -488,7 +488,7 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
                                   // config in the hot path.
   ServerConnectionPtr codec_;
   std::list<ActiveStreamPtr> streams_;
-  Stats::TimespanPtr conn_length_;
+  std::unique_ptr<Stats::Timespan<std::chrono::milliseconds>> conn_length_;
```

Comment:
Can we use `Stats::CompletableTimespanPtr` here instead?

Reply:
I think we're good once you address this one.

Reply:
@venilnoronha Actual object held by conn_length is later passed as a reference in UserAgent::completeConnectionLength(Stats::Timespan<std::chrono::milliseconds>& span) where the getRawDuration() method is used which is not present in CompletableTimespan.  That is the reason I have to use explicit type here rather than Stats::CompletableTimespanPtr.

Reply:
Got it.

Reply:
Thanks @venilnoronha for reviewing this change.